src/Login.ts

```diff
@@ -151,6 +179,36 @@ export default class Login {
     }
 }
 
+export interface OidcNativeFlow extends ILoginFlow {
+    type: "oidcNativeFlow";
+    clientId: string;
+}
+/**
+ * Finds static clientId for configured issuer, or attempts dynamic registration with the OP
+ * @param delegatedAuthConfig  Auth config from ValidatedServerConfig
+ * @param clientName Client name to register with the OP, eg 'Element', used during client registration with OP
+ * @param staticOidcClients static client config from config.json, used during client registration with OP
+ * @returns Promise<OidcNativeFlow> when oidc native authentication flow is supported and correctly configured
+ * @throws when oidc native flow is not configured, client can't register with OP, or any unexpected error
+ */
+const tryInitOidcNativeFlow = async (
+    delegatedAuthConfig: ValidatedServerConfig["delegatedAuthentication"] | undefined,
+    brand: string,
+    oidcStaticClients?: IConfigOptions["oidc_static_clients"],
+): Promise<OidcNativeFlow> => {
+    if (!delegatedAuthConfig) {
```

Comment:
I think I'd find it more intuitive if this check was hoisted to `getFlows` rather than throwing an exception which will immediately be caught. YMMV though.